bytes/bytes_pool.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gxbytes
+
+import (
+	"sync"
+)
+
+// BytesPool hold specific size []byte
+type BytesPool struct {
+	sizes  []int // sizes declare the cap of each slot
+	slots  []sync.Pool
+	length int
+}
+
+var defaultBytesPool = NewBytesPool([]int{16, 1 << 10, 2 << 10, 4 << 10, 8 << 10, 16 << 10, 32 << 10, 64 << 10})
```

Comment:
I was going to optimize for small object. now I found it's not necessary for dubbo-go

Reply:
0.5k ~ 64k is enough, but I found a 64k+ size in https://github.com/apache/dubbo-go-hessian2/blob/master/binary.go#L137, maybe it is a bug ?

Reply:
it just apply a big enough bytes, but it is also possible exceed.
i just realize that when u release back the bytes, u must rest the bytes, and check its cap, 
reject the bytes if the cap is not expected.

Reply:
if size is too large, it will cost so many cpu and heap:

```
BenchmarkAcquireBytesSize10k-8   	15934426	        71.8 ns/op	      32 B/op	       1 allocs/op
BenchmarkAcquireBytesSize60k-8   	16641849	        78.3 ns/op	      32 B/op	       1 allocs/op
BenchmarkAcquireBytesSize70k-8   	  182911	      6195 ns/op	   73728 B/op	       1 allocs/op
```

Reply:
> i just realize that when u release back the bytes, u must rest the bytes, and check its cap,
> reject the bytes if the cap is not expected.

yes, I had ignore if the cap is exceed

Reply:
ok， wait for your update